drivers/i2c/i2c_andes_atciic100.c

```diff
@@ -0,0 +1,983 @@
+/*
+ * Copyright (c) 2022 Andes Technology Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file I2C driver for AndesTech atciic100 IP
+ */
+
+#include <drivers/i2c.h>
+#include "i2c_andes_atciic100.h"
+
+#define DT_DRV_COMPAT andestech_atciic100
+
+struct i2c_atciic100_device_config {
+	uint32_t		i2c_base_addr;
+	uint32_t		i2c_irq_num;
+	struct _i2c_info	*i2c_info;
+};
+
+static int i2c_atciic100_master_send(const struct device *dev,
+	uint16_t addr, const uint8_t *data, uint32_t num, bool phase_stop);
+static int i2c_atciic100_master_receive(const struct device *dev,
+	uint16_t addr, uint8_t *data, uint32_t num, bool phase_stop);
+static int i2c_atciic100_slave_send(const struct device *dev,
+	const uint8_t *data, uint32_t num);
+static int i2c_atciic100_slave_receive(const struct device *dev,
+	uint8_t *data, uint32_t num);
+static void i2c_master_fifo_write(const struct device *dev, uint8_t is_init);
+static void i2c_master_fifo_read(const struct device *dev);
+static void i2c_slave_fifo_read(const struct device *dev, uint8_t is_fifo_full);
+static void i2c_slave_fifo_write(const struct device *dev);
+static void i2c_slave_addr_hit_handler(const struct device *dev);
+
+static int i2c_atciic100_init(const struct device *dev);
+
+const struct i2c_slave_callbacks *slave_callbacks;
+struct i2c_slave_config *slave_config;
+
+static void i2c_atciic100_default_control(const struct device *dev)
+{
+	const struct i2c_atciic100_device_config *dev_cfg = DEV_I2C_CFG(dev);
+	struct _i2c_info *i2c_info = dev_cfg->i2c_info;
+	uint32_t reg = 0;
+
+	k_sem_init(&i2c_info->i2c_transfer_sem, 0, 1);
+	k_sem_init(&i2c_info->i2c_busy_sem, 0, 1);
+	k_sem_give(&i2c_info->i2c_transfer_sem);
+	k_sem_give(&i2c_info->i2c_busy_sem);
+
+	/* Reset I2C bus */
+	reg = INWORD(I2C_CMD(dev));
+	reg &= (~CMD_MSK);
+	reg |= (CMD_RESET_I2C);
+	OUTWORD(I2C_CMD(dev), reg);
+
+	/* I2C query FIFO depth */
+	reg = INWORD(I2C_CFG(dev));
+	switch (reg & 0x3) {
+	case 0x0:
+		i2c_info->fifo_depth = 2;
+		break;
+	case 0x1:
+		i2c_info->fifo_depth = 4;
+		break;
+	case 0x2:
+		i2c_info->fifo_depth = 8;
+		break;
+	case 0x3:
+		i2c_info->fifo_depth = 16;
+		break;
+	}
+
+	/*
+	 * I2C setting: slave mode(default), standard speed
+	 * 7-bit, CPU mode
+	 */
+	OUTWORD(I2C_SET(dev), 0x0);
```

Comment:
you can likely use the sys_write32/sys_read32 in place of outword/inword macros here